Views/Widget-ResponsiveImage.liquid

```diff
@@ -1,12 +1,38 @@
-{% assign linkTo = Model.ContentItem.Content.ResponsiveImage.LinkTo.Text %}
 {% assign image = Model.Content.ResponsiveImage-Image %}
 
-{% if linkTo != nil %}
-    <a class="image image--link image--responsive {{ Model.ContentItem | animation_css }}" href="{{ linkTo }}" style="{{ Model.ContentItem | animation_styles }}" {{ Model.ContentItem | animation_data_attributes }}>
+{% assign linkUrl = Model.ContentItem.Content.ResponsiveImage.LinkTo.Text %}
+
+{% if Model.ContentItem.Content.LinkDestinationPart.DestinationUrl.Text != blank %}
+    {% assign linkUrl = Model.ContentItem.Content.LinkDestinationPart.DestinationUrl.Text %}
+{% endif %}
+
+{% assign openIn = Model.ContentItem.Content.LinkBehaviourPart.OpenIn.Text %}
+{% assign isModal = openIn == 'modal' or openIn == 'modal-wide' %}
+{% assign onClickEvent = Model.ContentItem.Content.LinkBehaviourPart.ClickEvent.Value %}
+
+{% if isModal %}
+    {% assign cssClasses = cssClasses | append: ' js-iframe-modal' %}
+{% endif %}
+
+{% if linkUrl != blank %}
+
+    <a href="{{ linkUrl }}"

```

Comment:
Not a dealbreaker, but in this instance we could use an approach similar to the Image widget and have a {{tag}} variable rather than needing to have the repeated outputs across two halves of an if statement. Where possible I think that's my preference, although there are scenarios (like heading) where an if statement is necessary.

Reply:
Good thinking.